### PR TITLE
G.Abstract Factory

### PR DESCRIPTION
**Pull request by M0narc**
Topic Abstract* factory.

### CODE REVIEW

File: Globant Abstract factory/src/com/globant/FactoryProducer.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.globant;
+
+public class FactoryProducer {
+    public static AbstractFactory getFactory (boolean Sql){
```
Comment (LeoBur) [inline Globant Abstract factory/src/com/globant/FactoryProducer.java]: Variables should begin with lower case

File: Globant Abstract factory/src/com/globant/NoSQLFactory.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.globant;
+
+public class NoSQLFactory extends AbstractFactory {
+
+    @Override
+    public Connection getConnection(String connectionType) {
+        if (connectionType.equalsIgnoreCase("MYSQL")) {
```
Comment (LeoBur) [inline Globant Abstract factory/src/com/globant/NoSQLFactory.java]: Change the condition order, it should be
```suggestion
        if ("MYSQL".equalsIgnoreCase(connectionType)) {
```
Also, if you are getting MYSQL, why are you returning a NoSql connection?

**Comment by M0narc** [comment]
Oh you're right about the lower case variable I will change it asap.
 I'll also fix the SQL example, I'm  sure
I had a reason for returning the Nosql but since I didn't left annotations
I don't quite remember why, I'll start using proper annotations from on.

File: Globant Abstract factory/src/com/globant/NoSQLFactory.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.globant;
+
+public class NoSQLFactory extends AbstractFactory {
+
+    @Override
+    public Connection getConnection(String connectionType) {
+        if ("MYSQL".equalsIgnoreCase(connectionType)) {
```
Comment (LeoBur) [inline Globant Abstract factory/src/com/globant/NoSQLFactory.java]: This doesn't seems to be right

File: Globant Abstract factory/src/com/globant/SQLFactory.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.globant;
+
+public class SQLFactory extends AbstractFactory{
+
+    @Override
+    public Connection getConnection(String connectionType) {
+        if(connectionType.equalsIgnoreCase("MYSQL")) {
```
Comment (LeoBur) [inline Globant Abstract factory/src/com/globant/SQLFactory.java]: Use yoda conditions constant part to the left
"MYSQL".equalsIgnoreCase()

File: Globant Abstract factory/src/com/globant/NoSQLFactory.java (hunk 3 above)
Comment (M0narc) [reply Globant Abstract factory/src/com/globant/NoSQLFactory.java]: The other day i clicked the "commit changes" button where you recomended a change in the code, this was a mistake, i'll fix it right now.

**Pull request closed without merging**